java/arcs/core/host/ParticleContext.kt

```diff
@@ -52,7 +52,15 @@ class ParticleContext(
     private val desyncedHandles: MutableSet<Handle> = mutableSetOf()
 
     // One-shot callback used to notify the arc host that the particle is in the Running state.
+    // If the particle is already in the running state, the method will be called, but not set.
     private var notifyReady: ((Particle) -> Unit)? = null
+        set(value) {
+            if (particleState == ParticleState.Running) {
+                value?.invoke(this.particle)
+            } else {
+                field = value
+            }
+        }
```

Comment:
Just double check, this will be called in a single-threaded fashion, right?

Reply:
Yes